Rename post scripts

WalterKolczynski-NOAA opened the pull request:
**Description**
Rename post scripts to remove the unnecessary "ncep" qualifier. Also changes the executable name for upp to be the one produced by upp (`upp.x`).

The ush/global_nceppost.sh script is removed as it is not used by any job. We may want to rename the gfs_post.sh script (the one that is used) to global_post.sh to be more reflective of its usage.

**Type of change**
- [x] Maintenance (alteration that does not change functionality)

**How Has This Been Tested?**
- [x] Forecast-only test on Hera
- [x] Cycled test on Hera
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

WalterKolczynski-NOAA committed:
Rename post scripts

Rename post scripts to remove the unnecessary "ncep" qualifier.

WalterKolczynski-NOAA committed:
Linter fixes for post rename

KateFriedman-NOAA reviewed (approved):
Changes look good! Thanks @WalterKolczynski-NOAA ! Approve pending resolution of linter warnings.

WalterKolczynski-NOAA committed:
Linter corrections for post j-job

aerorahul commented on jobs/JGLOBAL_ATMOS_POST:
```diff
@@ -1,46 +1,51 @@
+    . "${config_path}/config.${config}"
     status=$?
-    [[ $status -ne 0 ]] && exit $status
+    (( status != 0 )) && exit "${status}"
```
Are you sure `(( )) && ` or `(( )) || ` works?
My understanding is that `(( ))` is used when performing arithmetic.  The result of the arithmetic will set the exit code (0 for true).

WalterKolczynski-NOAA replied:
Yes. (( )) evaluates in arithmetic context and can handle inequalities: https://mywiki.wooledge.org/ArithmeticExpression

Quick test:
```
hfe02 rename_post/ > if ((  2 > 0 )); then echo "hi"; fi
hi
hfe02 rename_post/ > if ((  0 > 2 )); then echo "hi"; fi                                                                                                  
hfe02 rename_post/ > (( 2 > 0 )) || echo "hi"
hfe02 rename_post/ > (( 0 > 2 )) || echo "hi"                                                                                                             
hi
```

aerorahul replied:
Are we testing here if the arithmetic works or evaluating the expression?
```
(( $status -ne 0 ))
```
returns the status of the arithmetic evaluation of the expression.  So even if the arithmetic was wrong, but evaluated correctly, the return status would be True.
```
[[ $status -ne 0 ]]
```
evaluates the expression and returns a True or False.

In other words, just because it works, is this the right test?

WalterKolczynski-NOAA replied:
Guess I'm not sure I see the difference unless there is a string where a number should be (since a string would be treated as 0 in an arithmetic context).

aerorahul replied:
I suppose it doesn't make a difference in the end.  Just be consistent and make it a rule.
```
(( 2*0 )) && echo "hi"
(( 2*1 )) && echo "hi"
hi
```
Not sure why the first is `False`.  It is a perfectly fine arithmetic expression.

WalterKolczynski-NOAA replied:
I would only use this form as a logical unless it includes a comparison operator like the examples I gave. Really, we would never use `(( var = a + b ))` for math either, as it will return a non-zero exit code most of the time (use `var=$(( a + b ))` instead).

aerorahul commented on jobs/JGLOBAL_ATMOS_POST:
```diff
@@ -1,46 +1,51 @@
+. ${HOMEgfs}/env/${machine}.env post
 status=$?
-[[ $status -ne 0 ]] && exit $status
+(( status != 0 )) && exit "${status}"
```
same question as above.

aerorahul commented on jobs/JGLOBAL_ATMOS_POST:
```diff
@@ -1,46 +1,51 @@
 # Source machine runtime environment
 ##########################################
-. $HOMEgfs/env/${machine}.env post
+# shellcheck source-path=env
```
are we adding this now to every file?
Can this be set in higher level env?

WalterKolczynski-NOAA replied:
No, but we may be able to do without it. I was hoping to get fewer other complaints if shellcheck can source the right files, and I'm not sure it is even helping (especially this one, which I don't think is followed).

aerorahul commented on jobs/JGLOBAL_ATMOS_POST:
```diff
@@ -1,46 +1,51 @@
-export cycle="t${cyc}z"
+export cycle="t${cyc:?}z"
 setpdy.sh
+# shellcheck disable=SC1091
```
I am not sure littering the code w/ this type of stuff is a good idea.
Its distracting for one, and its hard to know what are we disabling and why specifically here.

WalterKolczynski-NOAA replied:
The latter issue can be solved by explaining in comment, but I was trying to make them minimally intrusive. So far, I've only needed shellcheck directives in a couple places per file. We could turn certain warnings off all the time, but, well, then they'd be off all the time.

aerorahul commented on jobs/JGLOBAL_ATMOS_POST:
```diff
@@ -68,30 +73,32 @@ export COMPONENT=${COMPONENT:-atmos}
 
-export COMIN=${COMIN:-$ROTDIR/$RUN.$PDY/$cyc/$COMPONENT}
-export COMOUT=${COMOUT:-$ROTDIR/$RUN.$PDY/$cyc/$COMPONENT}
+export COMIN=${COMIN:-${ROTDIR}/${RUN}.${PDY}/${cyc}/${COMPONENT}}
```
Is there ever a reason for `COMPONENT` in this script to not be `atmos`?
Line 70 should be:
`export COMPONENT="atmos"`

WalterKolczynski-NOAA replied:
Not that I know of. I was just updating for the linter.

aerorahul commented on jobs/JGLOBAL_ATMOS_POST:
```diff
@@ -117,9 +124,9 @@ export SLEEP_INT=5
 # Run relevant exglobal script
 
-$SCRgfs/ex${RUN}_atmos_nceppost.sh
+"${SCRgfs}/ex${RUN}_atmos_post.sh"
```
Are the `exgfs_atmos_post.sh` and `exgdas_atmos_post.sh` that different?
Just curious here.

WalterKolczynski-NOAA replied:
Not too different. Comparing more carefully and merging them is part of the next phase. This PR is supposed to just be renaming the scripts (and then all the linter complaints).

aerorahul reviewed (commented):
Several comments:
- Use of `(( ))` at status checks.  See specific comment
- Use of shellcheck statements.  See specific comment
- Some other, inline.

WalterKolczynski-NOAA committed:
Linter corrections for gdas post exscript

WalterKolczynski-NOAA committed:
Linter corrections for gfs post exscript

WalterKolczynski-NOAA committed:
Disable variable assigned but not defined shellcheck

Disables SC2154 (variable assigned but not defined).

WalterKolczynski-NOAA committed:
Remove excess default variable setting

With SC2154 turned off, the excessive usage of :? can be removed.

WalterKolczynski-NOAA committed:
Linter corrections for gfs_pos

WalterKolczynski-NOAA committed:
Last linter cleanup for post rename

WalterKolczynski-NOAA committed:
Disable shellcheck on sourced scripts

Turns off shellcheck parsing sourced scripts and removes some shellcheck
directives that are no longer necessary.

WalterKolczynski-NOAA committed:
Disable more shellcheck errors

WalterKolczynski-NOAA committed:
Suppress possible typo warning on NDATE

WalterKolczynski-NOAA committed:
Fix decimal treated as octal issue in post

After linter modifications, a variable was being treated as an octal, resulting
in the wrong result when adding.
